src/toil/job.py

```diff
@@ -2470,15 +2471,15 @@ def run(self, fileStore):
         # Assumes promises are fulfilled when parent job is run
         self.evaluatePromisedRequirements()
         userFunction = self._getUserFunction()
-        return self.addChildFn(userFunction, *self._args, **self._kwargs).rv()
+        return self.addChildFn(userFunction, *self._args, **self.userFunctionKwargs).rv()
```

Comment:
yeah that was my idea. We can rename it. Basically we need this job to have low requirements while retaining a copy of the user specified ones for the real user function